lib/Dialect/Secret/Transforms/ImportExecutionResult.cpp

```diff
@@ -0,0 +1,121 @@
+#include "lib/Dialect/Secret/Transforms/ImportExecutionResult.h"
+
+#include <fstream>
+#include <string>
+
+#include "lib/Dialect/Secret/IR/SecretOps.h"
+#include "llvm/include/llvm/ADT/STLExtras.h"    // from @llvm-project
+#include "mlir/include/mlir/IR/Attributes.h"    // from @llvm-project
+#include "mlir/include/mlir/IR/BuiltinTypes.h"  // from @llvm-project
+#include "mlir/include/mlir/IR/Operation.h"     // from @llvm-project
+#include "mlir/include/mlir/IR/Types.h"         // from @llvm-project
+#include "mlir/include/mlir/Support/LLVM.h"     // from @llvm-project
+
+namespace mlir {
+namespace heir {
+namespace secret {
+
+// this is in correspondence with insertExternalCall in AddDebugPort.cpp
+LogicalResult annotateResult(secret::GenericOp op,
+                             const std::vector<std::vector<double>> &data) {
+  auto getArrayOfDoubleAttr = [&](const std::vector<double> &row) {
+    auto type = Float64Type::get(op.getContext());
+    SmallVector<Attribute> elements =
+        llvm::to_vector<4>(llvm::map_range(row, [&](double value) -> Attribute {
+          return mlir::FloatAttr::get(type, value);
+        }));
+    return ArrayAttr::get(op.getContext(), elements);
+  };
+
+  int lineCount = 0;
+  auto tryGetLine = [&]() -> Attribute {
+    if (lineCount >= data.size()) {
+      return Attribute();
+    }
+    return getArrayOfDoubleAttr(data[lineCount++]);
+  };
+
+  // insert for each argument
+  for (auto i = 0; i != op.getBody()->getNumArguments(); ++i) {
+    auto lineAttr = tryGetLine();
+    if (!lineAttr) {
+      op.emitError("Not enough data for argument " + std::to_string(i));
+      return failure();
+    }
+    op.setOperandAttr(i, "plaintext.result", lineAttr);
```

Comment:
```suggestion
    op.setOperandAttr(i, "plaintext.result", lineAttr);
```

Can we name this so that the `secret` dialect is the actual prefix? Then people who see it will know to look in the `secret` dialect for it.

And put it as a dialect-level constant similar to the other attribute names.